File: app/src/main/java/agersant/polaris/features/browse/BrowseViewContent.java
package agersant.polaris.features.browse;

import android.content.Context;
import android.widget.FrameLayout;

import com.orangegangsters.github.swipyrefreshlayout.library.SwipyRefreshLayout;

import java.util.List;

import agersant.polaris.CollectionItem;


abstract class BrowseViewContent extends FrameLayout {

    public BrowseViewContent(Context context) {
        super(context);
    }

    abstract void updateItems(List<CollectionItem> items);

    void setOnRefreshListener(SwipyRefreshLayout.OnRefreshListener listener) {
    }

    int getScrollPosition() {
        return 0;
    }

    void setScrollPosition(int position) {
    }
}

File: app/src/main/java/agersant/polaris/features/browse/BrowseViewContent.kt
package agersant.polaris.features.browse

import agersant.polaris.CollectionItem
import android.content.Context
import android.widget.FrameLayout
import com.orangegangsters.github.swipyrefreshlayout.library.SwipyRefreshLayout.OnRefreshListener

internal abstract class BrowseViewContent(context: Context) : FrameLayout(context) {

    abstract fun updateItems(items: List<CollectionItem>)

    open fun setOnRefreshListener(listener: OnRefreshListener?) {}

    open fun saveScrollPosition() = 0

    open fun restoreScrollPosition(position: Int) {}
}
